CMakeLists.txt

```
@@ -650,4 +650,11 @@ if(MSVC)
 else()
   set(CPACK_GENERATOR TGZ)
 endif()
+
+if (APPLE)
+    set(CMAKE_INSTALL_RPATH "@executable_path/../lib")
+else(UNIX)
+    set(CMAKE_INSTALL_RPATH "$ORIGIN/../lib")
+endif()
```

Untested, but you have to keep in mind `CMAKE_INSTALL_LIBDIR` and `CMAKE_INSTALL_BINDIR` are configurable.
```suggestion
cmake_path(RELATIVE_PATH CMAKE_INSTALL_FULL_LIBDIR BASE_DIRECTORY "${CMAKE_INSTALL_FULL_BINDIR}" OUTPUT_VARIABLE install_rel_libdir)
if(APPLE)
    set(CMAKE_INSTALL_RPATH "@executable_path/${install_rel_libdir}")
else(UNIX)
    set(CMAKE_INSTALL_RPATH "$ORIGIN/${install_rel_libdir}")
endif()
```

---

You are 100% correct, thanks.